File: Programming Basics/Conditional Statements Advanced - Exercises/08.OnTimeForTheExam.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        int examHour = int.Parse(Console.ReadLine());
        int examMinute = int.Parse(Console.ReadLine());
        int arrivalHour = int.Parse(Console.ReadLine());
        int arrivalMinute = int.Parse(Console.ReadLine());

        TimeSpan examTime = new TimeSpan(examHour, examMinute, 0);
        TimeSpan arrivalTime = new TimeSpan(arrivalHour, arrivalMinute, 0);

        TimeSpan value = examTime.Subtract(arrivalTime);


        if (examTime == arrivalTime)
        {
            Console.WriteLine("On time");
            return;
        }
        if (examTime.Hours == arrivalTime.Hours && value.Minutes >= 1 && value.Minutes <= 30
            || examTime.Hours == arrivalTime.Hours && value.Minutes >= 1 && value.Minutes <= 30 && value.Hours >= 1
            || examTime.Hours != arrivalTime.Hours && value.Minutes >= 1 && value.Minutes <= 30 && value.Hours < 1)
        {
            Console.WriteLine("On time");
            Console.WriteLine($"{Math.Abs(value.Minutes)} minutes before the start");
        }
        else if (examTime.Hours == arrivalTime.Hours && value.Minutes > 30 && value.Minutes <= 59
            || examTime.Hours == arrivalTime.Hours && value.Minutes > 30 && value.Minutes <= 59 && value.Hours <= 1
            || examTime.Hours != arrivalTime.Hours && value.Minutes > 30 && value.Minutes <= 59 && value.Hours <= 1)
        {
            if (value.Hours >= 1)
            {
                Console.WriteLine("Early");
                Console.WriteLine($"{Math.Abs(value.Hours)}:{Math.Abs(value.Minutes):00} hours before the start");
            }
            else
            {
                Console.WriteLine("Early");
                Console.WriteLine($"{Math.Abs(value.Minutes)} minutes before the start");
            }
        }
        else if (examTime.Hours != arrivalTime.Hours && value.Hours >= 1)
        {
            Console.WriteLine("Early");
            Console.WriteLine($"{Math.Abs(value.Hours)}:{Math.Abs(value.Minutes):00} hours before the start");
        }
        else if (examTime.Hours == arrivalTime.Hours && arrivalTime.Minutes >= 1 && value.Minutes <= 59
            || examTime.Hours <= arrivalTime.Hours && value.Minutes >= 1 && value.Minutes <= 59
            || examTime.Minutes < arrivalTime.Minutes && value.Minutes >= 1 && value.Minutes <= 59)
        {
            Console.WriteLine("Late");
            Console.WriteLine($"{Math.Abs(value.Minutes)} minutes after the start");
        }
        else
        {
            Console.WriteLine("Late");
            if (value.Hours >= 1)
            {
                Console.WriteLine($"{Math.Abs(value.Minutes):00} minutes after the start");
            }
            else
            {
                if (value.Hours == 0)
                    Console.WriteLine($"{Math.Abs(value.Minutes):00} minutes after the start");
                else
                    Console.WriteLine($"{Math.Abs(value.Hours)}:{Math.Abs(value.Minutes):00} hours after the start");
            }
        }
    }
}

File: Programming Basics - Python/Conditional Statements Advanced - Exercises/08.OnTimeForTheExam.py
exam_hour = int(input())
exam_minute = int(input())
arrival_hour = int(input())
arrival_minute = int(input())

exam_time = exam_hour * 60 + exam_minute
arrival_time = arrival_hour * 60 + arrival_minute

time_difference = exam_time - arrival_time

if time_difference == 0:
    print('On time')
else:
    abs_difference = abs(time_difference)
    difference_hours = abs_difference // 60
    difference_minutes = abs_difference % 60

    if time_difference > 0:
        if abs_difference <= 30:
            print('On time')
            print(f'{difference_minutes} minutes before the start')
        else:
            print('Early')
            if difference_hours > 0:
                print(f'{difference_hours}:{difference_minutes:02d} hours before the start')
            else:
                print(f'{difference_minutes} minutes before the start')
    else:
        print('Late')
        if abs_difference < 60:
            print(f'{difference_minutes} minutes after the start')
        else:
            if difference_hours > 0:
                print(f'{difference_hours}:{difference_minutes:02d} hours after the start')
            else:
                print(f'{difference_minutes} minutes after the start')